raiden_libs/blockchain.py

```diff
@@ -53,37 +53,44 @@ def create_event_filter(
 
 
 def get_events(
-        w3: Web3,
+        web3: Web3,
         contract_manager: ContractManager,
         contract_name: str,
         event_name: str,
+        contract_address: str = None,
         from_block: Union[int, str] = 0,
         to_block: Union[int, str] = 'latest',
 ) -> List:
     """Returns events emmitted by a contract for a given event name, within a certain range.
 
     Args:
-        w3: A Web3 instance
+        web3: A Web3 instance
         contract_manager: A contract manager
         contract_name: The name of the contract
         event_name: The name of the event
+        contract_address: The address of the contract to be filtered, can be `None`
         from_block: The block to start search events
         to_block: The block to stop searching for events
 
     Returns:
-        All matching events"""
+        All matching events
+    """
+    filter_params = {
+        'fromBlock': from_block,
+        'toBlock': to_block,
+    }
+    if contract_address is not None:
+        filter_params['contract_address'] = contract_address
```

Comment:
maybe add something like `is_checksum_address()` validation here

Reply:
Added an 'assert` for now.